### PR TITLE
Migrate to Appyx for navigation

### PR DESCRIPTION
**Pull request by boswelja**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

* Removing `compose-destinations`, which was giving me anxiety attacks, and replace it with [Appyx](https://bumble-tech.github.io/appyx/).
* Transition animations are gone, we should look into something fancy to replace them.
* We will need to rethink our approach to hiding the navigation bar, there is no obvious way to check whether we have traversed a child node in Appyx.

**Tested changes:**

App still runs.

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

**Review by imashnake0 — CHANGES_REQUESTED** [review]
Great work @boswelja, let's bring changes to the layouts in another PR and keep this focused on specifically migrating to Appyx.

A couple things that we should probably fix:
- [ ] Padding and layout of the search bar is strange.
- [ ] Hiding the navigation bar on navigating to the media page.
- [ ] Navigation from the search bar.
- [ ] Profile icon is larger.
- [ ] Navigation to details page is glitchy.
- [ ] Layout changes (`Scaffold`, etc) in a different PR.

File: app/src/main/java/com/imashnake/animite/features/MainActivity.kt [hunk 1]
```diff
@@ -62,62 +25,13 @@ class MainActivity : ComponentActivity() {
...
-
-                val navController = rememberAnimatedNavController()
-                val navHostEngine = rememberAnimatedNavHostEngine(
-                    rootDefaultAnimations = RootNavGraphDefaultAnimations(
-                        enterTransition = { fadeIn(animationSpec = tween(1000)) },
-                        exitTransition = { fadeOut(animationSpec = tween(300)) },
-                    )
-                )
-
-                Box(Modifier.fillMaxSize()) {
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/MainActivity.kt:79]: I did the same work with 20% less lines of code, is that not convincing enough?

File: app/src/main/java/com/imashnake/animite/features/MainActivity.kt [hunk 2]
```diff
@@ -62,62 +25,13 @@ class MainActivity : ComponentActivity() {
...
-                        navGraph = NavGraphs.root,
-                        modifier = Modifier
-                            .align(Alignment.TopCenter)
-                            .background(MaterialTheme.colorScheme.background)
-                            .fillMaxSize(),
-                        navController = navController,
-                        engine = navHostEngine
-                    )
-
-                    val searchBarBottomPadding: Dp by animateDpAsState(
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/MainActivity.kt:90]: I tested it and it seems to be working fine, am I missing something?

File: app/src/main/java/com/imashnake/animite/features/RootNode.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                        )
+                        NavigationBarItem(
+                            selected = screenState.last().key.navTarget == NavTarget.Profile,
+                            onClick = { backStack.newRoot(NavTarget.Profile) },
+                            icon = { NavTarget.Profile.Icon() }
+                        )
+                    }
+                },
+                floatingActionButton = {
+                    SearchBar(onItemClicked = { /* TODO */ })
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/RootNode.kt:81]: Good catch, I forgot to add the nav call

File: app/src/main/java/com/imashnake/animite/features/RootNode.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                },
+                floatingActionButtonPosition = FabPosition.End
+            ) {
+                Children(
+                    navModel = backStack,
+                    modifier = Modifier.padding(it),
+                    transitionHandler = rememberBackstackFader()
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/RootNode.kt:88]: I noticed this too, it probably has something to do with the way details are loaded in the media page. I can start making changes there if you like

[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/RootNode.kt:88]: Probably related: https://github.com/bumble-tech/appyx/discussions/246

**Comment by boswelja** [comment]
Closing this due to blockers with Appyx

**Pull request closed without merging**